src/main/java/org/opentripplanner/routing/graph/Graph.java

```diff
@@ -215,20 +216,38 @@
     /** Parent stops **/
     public Map<FeedScopedId, Stop> parentStopById = new HashMap<>();
 
-    /** Data model for Raptor routing */
+    // TripPatterns used to be reached through hop edges, but we're not creating on-board transit vertices/edges anymore.
+    public Map<String, TripPattern> tripPatternForId = Maps.newHashMap();
+
+    /** Interlining relationships between trips. */
+    public final BiMap<Trip,Trip> interlinedTrips = HashBiMap.create();
+
+    /** Data model for Raptor routing, with realtime updates applied (if any). */
     public transient TransitLayer transitLayer;
 
+    /** Data model for Raptor routing, with realtime updates applied (if any). */
+    public transient TransitLayer realtimeTransitLayer;
+
+    // Hack. I've tried three different ways of generating unique labels.
+    // Previously we were just tolerating edge label collisions.
+    // For some reason we're repeatedly generating splits on the same edge objects, despite a comment that said it was
+    // guaranteed there would only ever be one split per edge. This is going to fail as soon as we load a base OSM graph
+    // and build transit on top of it.
+    public long nextSplitNumber = 0;
+
```

Comment:
If we are going to use sequential IDs, we need to somehow establish what the highest ID used is when a graph is deserialized. So although it's appealing to make this counter part of the splitter vertex class, it can't really be static (or if it is static, it needs to be reset from a non-static field on the serialized Graph, which is even more complicated).

I tried a couple of non-sequential-ID approaches that would generate semantic IDs based on the edges that were being split and the split location, and didn't manage to create such an ID that avoided duplicates. So to me this implies there is something wrong with the splitting strategy, that it may be producing a lot of unnecessary splits in the same locations.

In the longer term what I'd rather do is figure out why there would ever be more than one split for a given edge and coordinate, eliminate that problem, and then give the splits non-sequential IDs. So I see this nextSplitNumber field as a stopgap.

Reply:
Sounds reasonable. We can leave this for now and I am sure we will come back to it - hopefully finding a better strategy on splitting edges.